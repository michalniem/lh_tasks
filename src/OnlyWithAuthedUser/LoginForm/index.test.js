import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import { LoginForm } from "./index";

describe("LoginForm", () => {
  const wrapper = shallow(<LoginForm />);

  test("should render correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("should set proper input value after input change", () => {
    const emailInput = wrapper.find("#email");
    const passwordInput = wrapper.find("#password");

    emailInput.simulate("change", {
      target: { name: "email", value: "example@example.com" },
    });
    passwordInput.simulate("change", {
      target: { name: "password", value: "examplePassword" },
    });

    expect(emailInput.get(0).props.value).toEqual("example@example.com");
    expect(passwordInput.get(0).props.value).toEqual("examplePassword");
  });

  // test("should set width style based on progress state", () => {
  //   const progressBarStyle = wrapper
  //     .find("[data-test-id='progressBar']")
  //     .prop("style");
  //   expect(progressBarStyle).toHaveProperty("width", "50%");
  // });

  // test("should render correctly", () => {
  //   expect(toJson(wrapper)).toMatchSnapshot();
  // });
});
