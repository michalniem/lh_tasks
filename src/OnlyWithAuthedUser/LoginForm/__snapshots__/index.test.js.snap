// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    className="form__label"
    htmlFor="email"
  >
    email
  </label>
  <input
    className="form__input"
    id="email"
    name="email"
    onChange={[Function]}
    type="email"
    value=""
  />
  <label
    className="form__label"
    htmlFor="password"
  >
    password
  </label>
  <input
    className="form__input"
    id="password"
    name="password"
    onChange={[Function]}
    type="password"
    value=""
  />
  <button
    type="submit"
  >
    Login
  </button>
</form>
`;
